utils/tolerance.stanza

```diff
@@ -0,0 +1,60 @@
+#use-added-syntax(jitx)
+defpackage ocdb/tolerance:
+  import core
+  import collections
+  import math
+
+; A generic value with some tolerance. May be offcenter (upper tolerance != lower tolerance)
+public pcb-struct ocdb/tolerance/Toleranced:
+  typ: Double,
+  tol+:Double,
+  tol-:Double,
+
+defmethod print (o:OutputStream, t:Toleranced):
+  println(o, "Toleranced(%_ <= %_ <= %_)" % [min-value(t), typ(t), max-value(t)])
+
+; Create a tolerance with minimum, typical, and maximum
+public defn min-typ-max (min:Double, typ:Double, max:Double) -> Toleranced:
+  Toleranced(typ, typ - min, max - typ)
+
+; Create a tolerance from percentages higher and lower
+public defn tol% (tol+:Double, typ:Double, tol-:Double) -> Toleranced:
+  val plus = tol+ * typ
+  val minus = tol- * typ
+  Toleranced(typ, plus, minus)
+
+; Create a tolerance from differences higher and lower
+public defn tol (typ:Double, tol+:Double, tol-:Double) -> Toleranced:
+  Toleranced(typ, tol+, tol-)
+
+; Create a tolerance with the same upper and lower tolerances
+public defn tol (typ:Double, tol+-:Double) -> Toleranced:
+  tol(typ, tol+-, tol+-)
+
+; Return the upper tolerance, as a percentage
+public defn tol+% (t:Toleranced) -> Double:
+  tol+(t) / typ(t)
```

Comment:
careful if `typ(t)` equals `0.0`

Reply:
I think if `typ == 0.0` then this is still valid, the result would be NaN

Reply:
ok, I would like to see this edge case covered in the unit tests

Reply:
Better to crash with `fatal` than to return NaN.

Reply:
It actually returns infinity in this case. Should we still crash?

Reply:
Yes. Crashing is good. I cannot imagine a scenario where a user will rely upon the +Inf for his code to work.

Reply:
Fixed, these functions now call `fatal`.